### ISSUE TITLE
@Mapper failed for properties starting with set

### ISSUE THREAD

**Issue by tiagolabsit** [mapstruct/mapstruct#2152]
When my source have a property starting with set.., some thing goes wrong..

**Resource:**
```
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class TestResource {

    private String setor;

}
```

**Entity:**
```
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class TestEntity {

    private String setor;

}
```

**Mapper:**
```
import br.com.contmatic.gestaosst.canonical.schema.v1.TestResource;
import br.com.contmatic.gestaosst.model.TestEntity;
import org.mapstruct.Mapper;

@Mapper(componentModel = "spring")
public interface TestMapper {

    TestEntity toEntity(TestResource resource);

    TestResource toResource(TestEntity entity);

}
```

**And the gerenated code:** (where is my property setor?)
```
import br.com.contmatic.gestaosst.canonical.schema.v1.TestResource;
import br.com.contmatic.gestaosst.canonical.schema.v1.TestResource.TestResourceBuilder;
import br.com.contmatic.gestaosst.model.TestEntity;
import br.com.contmatic.gestaosst.model.TestEntity.TestEntityBuilder;
import javax.annotation.Generated;
import org.springframework.stereotype.Component;

@Generated(
    value = "org.mapstruct.ap.MappingProcessor",
    date = "2020-07-15T20:41:55-0300",
    comments = "version: 1.3.0.Final, compiler: javac, environment: Java 1.8.0_231 (Oracle Corporation)"
)
@Component
public class TestMapperImpl implements TestMapper {

    @Override
    public TestEntity toEntity(TestResource resource) {
        if ( resource == null ) {
            return null;
        }

        TestEntityBuilder testEntity = TestEntity.builder();

        return testEntity.build();
    }

    @Override
    public TestResource toResource(TestEntity entity) {
        if ( entity == null ) {
            return null;
        }

        TestResourceBuilder testResource = TestResource.builder();

        return testResource.build();
    }
}
```

**The best workarround:** (called gambiarra in Brasil)
```
import br.com.contmatic.gestaosst.canonical.schema.v1.TestResource;
import br.com.contmatic.gestaosst.model.TestEntity;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.Mappings;

@Mapper(componentModel = "spring")
public interface TestMapper {

    @Mappings(@Mapping(source = "setor", target = "or"))
    TestEntity toEntity(TestResource resource);

    @Mappings(@Mapping(source = "setor", target = "or"))
    TestResource toResource(TestEntity entity);

}
```

**The new generated code:** (Oh my beautiful field!)
```
import br.com.contmatic.gestaosst.canonical.schema.v1.TestResource;
import br.com.contmatic.gestaosst.canonical.schema.v1.TestResource.TestResourceBuilder;
import br.com.contmatic.gestaosst.model.TestEntity;
import br.com.contmatic.gestaosst.model.TestEntity.TestEntityBuilder;
import javax.annotation.Generated;
import org.springframework.stereotype.Component;

@Generated(
    value = "org.mapstruct.ap.MappingProcessor",
    date = "2020-07-15T20:49:17-0300",
    comments = "version: 1.3.0.Final, compiler: javac, environment: Java 1.8.0_231 (Oracle Corporation)"
)
@Component
public class TestMapperImpl implements TestMapper {

    @Override
    public TestEntity toEntity(TestResource resource) {
        if ( resource == null ) {
            return null;
        }

        TestEntityBuilder testEntity = TestEntity.builder();

        testEntity.setor( resource.getSetor() );

        return testEntity.build();
    }

    @Override
    public TestResource toResource(TestEntity entity) {
        if ( entity == null ) {
            return null;
        }

        TestResourceBuilder testResource = TestResource.builder();

        testResource.setor( entity.getSetor() );

        return testResource.build();
    }
}
```

**Comment by tiagolabsit** [comment]
I identified this problem is related with @Builder or @SuperBuidler annotation.

**Comment by filiphr** [comment]
This is similar to #1799. However, that should be fixed in 1.3.1.Final. Most likely it is related to `@SuperBuilder`. It seems like we don't treat the `setor` as a fluent setter.

Can you please share the signature of that method in the `TestResourceBuilder`?

**Comment by tiagolabsit** [comment]
The code is generated by lombok, `@SuperBuilder` is a `@Builder` with superclasses.
This is the docs:
https://projectlombok.org/features/Builder
https://projectlombok.org/features/experimental/SuperBuilder

And its the same problem of issue #1799

**Comment by filiphr** [comment]
It is similar to #1799, it can't be the same since that one is fixed and working properly on 1.3.1.Final.

I am not a Lombok user and in the documentation it is not explained how the generated code looks like. We at MapStruct rely on the AST (Abstract Syntax Tree) from the compiler API to detect whether a method is a fluent setter or not. It can be related to Lombok not exposing that or the signature is different. Therefore, I would really appreciate if you can delombok your class and send us the method signature so that we can try it out in a setting without Lombok.

**Comment by tiagolabsit** [comment]
@filiphr thanks, i update the dependecy version and all goes well!!
